collections/associate_with_test.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../testing/asserts.ts";
+import { associateWith } from "./associate_with.ts";
+
```

Comment:
We should test for consistent behaviour if the same key appears mumtiple times in the input (see my comment on docs)

Reply:
Fixed, @LionC can you please review.